pkg/deploy/deploy_test.go

```diff
@@ -109,3 +116,97 @@ func TestGetParameters(t *testing.T) {
 		})
 	}
 }
+
+func TestRemoveFailedScaleset(t *testing.T) {
+	ctx := context.Background()
+	rgName := "testRg"
+	vmssName := "testVMSS"
+	for _, tt := range []struct {
+		name  string
+		mocks func(*mock_compute.MockVirtualMachineScaleSetsClient)
+		want  bool
+	}{
+		{
+			name: "VMSS not found",
+			mocks: func(vmss *mock_compute.MockVirtualMachineScaleSetsClient) {
+				vmss.EXPECT().Get(ctx, rgName, vmssName).Return(
+					mgmtcompute.VirtualMachineScaleSet{},
+					autorest.DetailedError{
+						StatusCode: http.StatusNotFound,
+					},
+				)
+
+			},
+			want: true,
+		},
+		{
+			name: "Found but not failed",
+			mocks: func(vmss *mock_compute.MockVirtualMachineScaleSetsClient) {
+				vmss.EXPECT().Get(ctx, rgName, vmssName).Return(
+					mgmtcompute.VirtualMachineScaleSet{
+						VirtualMachineScaleSetProperties: &mgmtcompute.VirtualMachineScaleSetProperties{
+							ProvisioningState: to.StringPtr(string(mgmtcompute.ProvisioningStateSucceeded)),
+						},
+					},
+					nil,
+				)
+			},
+			want: false,
+		},
+		{
+			name: "Found, failed, deletion failed",
+			mocks: func(vmss *mock_compute.MockVirtualMachineScaleSetsClient) {
+				vmss.EXPECT().Get(ctx, rgName, vmssName).Return(
+					mgmtcompute.VirtualMachineScaleSet{
+						Name: to.StringPtr(vmssName),
+						VirtualMachineScaleSetProperties: &mgmtcompute.VirtualMachineScaleSetProperties{
+							ProvisioningState: to.StringPtr(string(mgmtcompute.ProvisioningStateFailed)),
+						},
+					},
+					nil,
+				)
+				vmss.EXPECT().DeleteAndWait(ctx, rgName, vmssName).Return(
+					autorest.NewErrorWithError(nil, "Future", "WaitForCompletion", &http.Response{
+						StatusCode: http.StatusTeapot,
+					}, "the number of retries has been exceeded"),
+				)
```

Comment:
I thought it might be informative to those reading the code to see that it would be an error with the message, "the number of retries has been exceeded".

Reply:
took your suggestion and made it simpler